### ISSUE TITLE
Stdlib: Math

### ISSUE THREAD

**Issue by phated** [grain-lang/grain#251]
Once #250 is completed, we'll need an awesome math module in the stdlib to work with numbers.

**Comment by Ultra-Instinct-05** [comment]
@ospencer I was wondering If I could take a stab at this ? If you could share your thoughts on how you want this implemented (what functions and the details), I could do some research and try to implement this module.

**Comment by ospencer** [comment]
Absolutely! I'll ping you on Discord with some info.

**Comment by ospencer** [comment]
Functions to implement:
```
add // alias of (+)
sub // alias of (-)
mul // alias of (*)
div // alias of (/)
sqrt
min
max
ceil
floor
trunc
round
abs
neg
```

**Comment by phated** [comment]
This landed as `Number` in #687 🎉 Thanks @Ultra-Instinct-05 and @spotandjake for your work on it!
